### PR TITLE
[improve][branch-2.9]upgrade the bookkeeper version to 4.14.7

### PR DESCRIPTION
**Pull request by horizonzy**
Fixes #xyz

Master Issue: #xyz

PIP: #xyz 

### Motivation
BookKeeper 4.14.7 has been released and it contains a lot of notable bug fixes. We need to upgrade the BookKeeper dependency from 4.14.5 to 4.14.7.

BookKeeper release notes can be found:
https://github.com/apache/bookkeeper/releases

### Modifications
Upgrade BookKeeper dependency from 4.14.5 to 4.14.7

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by nicoloboschi — COMMENTED** [review]
please fix the license check

**Review by hangc0276 — CHANGES_REQUESTED** [review]
Please hold on to the upgrade due to the RocksDB issue.  https://github.com/apache/bookkeeper/issues/3734

**Comment by hangc0276** [comment]
BookKeeper 4.14.7 has been released, we can use BookKeeper 4.14.7 directly.

**Comment by hangc0276** [comment]
After this PR https://github.com/apache/pulsar/pull/19645 merged, I will rebase branch-2.9

**Comment by michaeljmarshall** [comment]
@horizonzy - will you please update the PR description to describe this change? Thanks!

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java [hunk 1]
```diff
@@ -35,6 +35,7 @@
 import io.netty.util.Recycler;
 import io.netty.util.Recycler.Handle;
 import java.time.Clock;
+import java.util.ArrayList;
```
Comment (codelipenghui) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:38]: Is it required?

Comment (codelipenghui) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:38]: If it fixed a build issue, I think we'd better have a separate PR.

Comment (hangc0276) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:38]: The branch keeps cherry-picking new PRs without building success. I have pushed one PR to fix the build, https://github.com/apache/pulsar/pull/19645, and the CI needs to wait for a long time and rerun a few times to pass.

Let me clean up the build-related changes to see whether we can build success.

Comment (hangc0276) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:38]: The branch-2.9 still build failed.
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project managed-ledger: Compilation failure
[ERROR] /Users/hangc/Workspace/sourcecode/CR/pulsar/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:[436,85] cannot find symbol
[ERROR]   symbol:   class ArrayList
[ERROR]   location: class org.apache.bookkeeper.mledger.impl.ManagedLedgerImpl

```

Comment (hangc0276) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:38]: Pushed one PR to fix it. https://github.com/apache/pulsar/pull/19669. After that PR is merged, I will rebase branch-2.9

**Review by maxsxu — APPROVED** [review]
Thanks! pulsar 2.9 endorses bookkeeper 4.14.7

**Review by coderzc — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**